File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.java
class Solution {
    public int maxProfit(int[] nums) {
        int cnt=0;
        for(int i=1;i<nums.length;i++){
            if(nums[i]>nums[i-1]){
                cnt+=(nums[i]-nums[i-1]);
            }
        }
        return cnt;
    }
}

File: Data Structures & Algorithms/decode-string/submission-0.cpp
class Solution {
public:
    string decodeString(string s) {
        int k = 0;
        string curr;
        stack<int> cst;
        stack<string> sst;

        for(char ch:s){
            if(isdigit(ch)){
                k = (k*10)+(ch-'0');
            }else if(ch == '['){
                cst.push(k);
                sst.push(curr);
                k = 0;
                curr.clear();
            }else if(ch == ']'){
                int cnt = cst.top();
                cst.pop();
                string prev = sst.top();
                sst.pop();
                for(int i=0;i<cnt;i++){
                    prev += curr;
                }
                curr = prev;
            }else{
                curr += ch;
            }
        }
        return curr;
    }
};